ntoskrnl/se/accesschk.c

```diff
@@ -394,13 +394,16 @@ SepAccessCheck(
     BOOLEAN Defaulted;
     NTSTATUS Status;
     PACCESS_TOKEN Token = NULL;
-    ACCESS_CHECK_RIGHTS AccessCheckRights = {0};
+    ACCESS_CHECK_RIGHTS AccessCheckRights;
 
     PAGED_CODE();
 
     /* A security descriptor must be expected for access checks */
     ASSERT(SecurityDescriptor);
 
+    /* Fill the whole access rights array with zeros */
```

Comment:
Not sure the comment is necessary, the code is already self-explainable.
But again, the structure is local to that function and used only within it, so the story of security and leaks is not that relevant here.
It would however be relevant, if somehow you then did a copy of the contents of this structure to an external buffer given as parameter to that SepAccessCheck function.

Reply:
Agreed, this isn't useful for this particular variable. If we want to preemptively zero stack space, we can do it, but we should let the compiler do it, no pollute our code: https://msrc.microsoft.com/blog/2020/05/solving-uninitialized-stack-memory-on-windows/